examples_tests/common/CommonAPI.h

```diff
@@ -293,6 +293,211 @@ class CommonAPI
+						outFamilyProp.graphics.supportsGraphics = hasGraphicsFlag;
+						outFamilyProp.graphics.supportsCompute = hasComputeFlag;
+						outFamilyProp.graphics.supportsTransfer = hasTransferFlag;
+						outFamilyProp.graphics.supportsSparseBinding = hasSparseBindingFlag;
+						outFamilyProp.graphics.supportsPresent = supportsPresent;
+						outFamilyProp.graphics.supportsProtected = hasProtectedFlag;
+					}
+					
+					// Present
+					if(supportsPresent && (outFamilyProp.present.index == QueueFamilyProps::InvalidIndex || outFamilyProp.present.supportsGraphics == false))
+					{
+						outFamilyProp.present.index = familyIndex;
+						outFamilyProp.present.supportsGraphics = hasGraphicsFlag;
+						outFamilyProp.present.supportsCompute = hasComputeFlag;
+						outFamilyProp.present.supportsTransfer = hasTransferFlag;
+						outFamilyProp.present.supportsSparseBinding = hasSparseBindingFlag;
+						outFamilyProp.present.supportsPresent = supportsPresent;
+						outFamilyProp.present.supportsProtected = hasProtectedFlag;
+					}
+
+					// Compute
+					if(hasComputeFlag && !hasGraphicsFlag && outFamilyProp.compute.index == QueueFamilyProps::InvalidIndex)
+					{
+						outFamilyProp.compute.index = familyIndex;
+						outFamilyProp.compute.supportsGraphics = hasGraphicsFlag;
+						outFamilyProp.compute.supportsCompute = hasComputeFlag;
+						outFamilyProp.compute.supportsTransfer = hasTransferFlag;
+						outFamilyProp.compute.supportsSparseBinding = hasSparseBindingFlag;
+						outFamilyProp.compute.supportsPresent = supportsPresent;
+						outFamilyProp.compute.supportsProtected = hasProtectedFlag;
+					}
+
+					// Transfer
+					if(hasTransferFlag && !hasGraphicsFlag && !hasComputeFlag && outFamilyProp.transfer.index == QueueFamilyProps::InvalidIndex)
+					{
+						outFamilyProp.transfer.index = familyIndex;
+						outFamilyProp.transfer.supportsGraphics = hasGraphicsFlag;
+						outFamilyProp.transfer.supportsCompute = hasComputeFlag;
+						outFamilyProp.transfer.supportsTransfer = hasTransferFlag;
+						outFamilyProp.transfer.supportsSparseBinding = hasSparseBindingFlag;
+						outFamilyProp.transfer.supportsPresent = supportsPresent;
+						outFamilyProp.transfer.supportsProtected = hasProtectedFlag;
+					}
+				}
+
+				// If Graphics supports Present, then use Graphics for Present
+				if(extractedInfo.queueFamilyProps.present.index != extractedInfo.queueFamilyProps.graphics.index && extractedInfo.queueFamilyProps.graphics.supportsPresent) 
+				{
+					extractedInfo.queueFamilyProps.present = extractedInfo.queueFamilyProps.graphics;
+				}
+
+				// If a unique Compute is not found but Graphics supports Compute, then use Graphics for Compute
+				if(extractedInfo.queueFamilyProps.compute.index == QueueFamilyProps::InvalidIndex && extractedInfo.queueFamilyProps.graphics.supportsCompute)
+				{
+					extractedInfo.queueFamilyProps.compute = extractedInfo.queueFamilyProps.graphics;
+				}
```

Comment:
its good to have graphics and compute be separate queues but probably not separate families

your function should know if you require graphics (headless compute) or not then pick a family that definitely has compute, but conditionally graphics

the fallback should be to have two separate families for compute and graphics (dunno if spec even allows for not having a family that doesnt support both compute and graphics)

Reply:
> its good to have graphics and compute be separate queues but probably not separate families

I will have compute and graphics be different queues (if possible and maxQueues allow)
I'm not sure about them being different queue families or not and I'm still researching this, but I assumed them being different families would have benefits and have some more concurrency or whatever xD

vulkan spec:
> If an implementation exposes any queue family that supports graphics operations, at least one queue family of at least one physical device exposed by the implementation must support both graphics and compute operations.

Reply:
> I'm not sure about them being different queue families or not and I'm still researching this, but I assumed them being different families would have benefits and have some more concurrency or whatever xD

I;ve never heard of having different families doing performance benefits.